src/main/java/com/facebook/openwifirrm/ucentral/UCentralClient.java

```diff
@@ -195,27 +202,117 @@ public boolean login() {
 		}
 
 		// Parse access token from response
-		JSONObject respBody;
+		WebTokenResult token;
 		try {
-			respBody = new JSONObject(response.getBody());
-		} catch (JSONException e) {
+			token = gson.fromJson(response.getBody(), WebTokenResult.class);
+		} catch (JsonSyntaxException e) {
 			logger.error("Login failed: Unexpected response", e);
 			logger.debug("Response body: {}", response.getBody());
 			return false;
 		}
-		if (!respBody.has("access_token")) {
+		if (token.access_token == null || token.access_token.isEmpty()) {
 			logger.error("Login failed: Missing access token");
-			logger.debug("Response body: {}", respBody.toString());
+			logger.debug("Response body: {}", token.toString());
 			return false;
 		}
-		this.accessToken = respBody.getString("access_token");
+		this.accessToken = token;
 		logger.info("Login successful as user: {}", username);
-		logger.debug("Access token: {}", accessToken);
+		logger.debug("Access token: {}", accessToken.access_token);
+		logger.debug("Refresh token: {}", accessToken.refresh_token);
 
 		// Load system endpoints
 		return loadSystemEndpoints();
 	}
 
+	/**
+	 * Check if an access token is expired.
+	 *
+	 * @return true if the access token is expired
+	 */
+	private boolean isAccessTokenExpired() {
+		return accessToken.created + accessToken.idle_timeout <
```

Comment:
what does idle timeout mean here? there's `expires_in` - why not use that?

Reply:
```
expires_in":2592000,"idle_timeout":300
```
This is printed by access token. I think `idle_timeout` makes more sense here. But open to any suggestion

Reply:
What I imagine "idle_timeout" to mean is that if there's no use of that token within that time period then the token is considered expired. However, if we take it to mean that this code is wrong since it never actually keeps track of last access time and uses a constant time of (creation + idle timeout).

Reply:
I think idle_timeout is important, i just don't know what it actually means in this context